### DISCUSSION TITLE
Add testers using fastlane to google play console for an app

### DISCUSSION THREAD

**Discussion by Teja25-web** [fastlane/fastlane#22197] [Q&A]
We are developing scripts to automate app publishing and few tasks around it. 
We want to add testers to an app on Google play console using Fastlane but couldn't find anything other than firebase_app_distribution plugin.
We want to avoid going this way as it involves adding their dependency to the application code base.
Is anyone aware of any way that could add testers to an android app without making any changes to an app.

**Comment by james598grose** [comment]
Hello, @Teja25-web 
You can use the fastlane supply command to add testers to your app on Google Play Console without modifying the app's codebase. Here's an example:
Set up your Google Play JSON key file (your-google-play-json-key.json).
Run the following command:
fastlane supply --json_key your-google-play-json-key.json --package_name your-app-package-name --track internal --add-testers --testers_email "tester1@example.com,tester2@example.com"
This command adds testers to the internal track without modifying the app's code.
Alternatively, you can use the Google Play Console API directly using the google-api-client gem. This approach requires more setup but gives you full control over the API calls.
Note: Ensure you have the correct permissions and credentials set up in your Google Play Console account.
[destinycard.com](https://www.mydestiny-card.com)


Best Regards,
james598grose

**Reply by Teja25-web** [reply]
Thanks @james598grose for you response. I did try the fastlane supply command but getting as "Invalid option: --add-testers". I tried to check this field in fastlane supply documentation and couldn't find it. Could you please help me with any link for this command.

**Comment by mataku** [comment]
@Teja25-web 
We could not do that because fastlane doesn't have actions using https://developers.google.com/android-publisher/api-ref/rest/v3/edits.testers, so need to add new actions using them 📝 

> Note: while it is possible in the Play Console UI to add testers via email lists in the Play Console UI, this resource does not support email lists.

As described in the above webpage, tester Web API has some limitations. Not sure if you could archive it, as the tester API has more limitations than on the browser.
